### PR TITLE
Nca

### PR DESCRIPTION
**Pull request by riveSunder**
add neural cellular automata simulator

### CODE REVIEW

File: coverage.txt [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
+yuca/ca/common.py                    124     26    79%
+yuca/ca/continuous.py                352     40    89%
+yuca/ca/neural.py                     74      1    99%
+yuca/configs.py                       89     21    76%
+yuca/kernels.py                      118     74    37%
 yuca/patterns.py                      59     43    27%
 yuca/utils.py                        121     47    61%
 ------------------------------------------------------
-TOTAL                               1417    657    54%
+TOTAL                               1564    378    76%
```
Comment (riveSunder) [inline coverage.txt:22]: Meets checklist for running tests before PRs and satisfies Issue #31

File: testing/yuca/ca/test_cca.py [hunk 2]
```diff
@@ -4,19 +4,50 @@
...
+            if frame_mode == 0:
+                self.assertEqual(1, len(frame_return))
+            elif True in (frame_mode == np.array([1,2])):
+                self.assertEqual(2, len(frame_return))
+            elif True in (frame_mode == np.array([3])):
+                self.assertEqual(3, len(frame_return))
+            elif True in (frame_mode == np.array([4])):
+                self.assertEqual(4, len(frame_return))
+            
+    def test_alive_maxk(self):
```
Comment (riveSunder) [inline testing/yuca/ca/test_cca.py:33]: typo, should be `test_alive_mask`

File: testing/yuca/ca/test_cca.py [hunk 3]
```diff
@@ -4,19 +4,50 @@
...
+        masked = ca(x)
+
+        
+
     def test_multiverse_forward(self):
 
-        for mode in ["functional", "neurofunctional", "neural"]:
-            ca = CA(ca_mode = mode)
-            ca.random_init()
+        for mode in ["functional", "neurofunctional"]:
```
Comment (riveSunder) [inline testing/yuca/ca/test_cca.py:45]: Neural functionality moved to `NCA` class. However, `ca_modes` should also be cleaned up before opening a PR from apprentice to master.

File: testing/yuca/test_configs.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import numpy as np
+
+import torch
+
+from yuca.ca.continuous import CCA
+from yuca.configs import get_smooth_life_config,\
+    get_orbium_config,\
+    get_geminium_config
+
+
```
Comment (riveSunder) [inline testing/yuca/test_configs.py:14]: Clean up extra white space.

File: yuca/ca/common.py [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+        get_gaussian_kernel, \
+        get_dogaussian_kernel, \
+        get_gaussian_edge_kernel, \
+        get_cosx2_kernel, \
+        get_dogaussian_edge_kernel
+         
+from yuca.patterns import get_orbium, \
+        get_smooth_puffer
+
+
```
Comment (riveSunder) [inline yuca/ca/common.py:37]: Clean up extra white space.

File: yuca/ca/continuous.py [hunk 6]
```diff
@@ -480,80 +355,6 @@ def add_identity_kernel(self, **kwargs):
 
         self.id_dim = dim_x
 
-
-    def initialize_id_layer(self):
```
Comment (riveSunder) [inline yuca/ca/continuous.py:484]: id and neighborhood layers moved to parent class.

File: yuca/ca/continuous.py [hunk 7]
```diff
@@ -768,119 +546,6 @@ def get_frame(self, universe, mode=0):
         elif mode == 4:
             return new_universe, universe, neighborhoods, update 
 
-    def fit_growth(self, target, **kwargs):
```
Comment (riveSunder) [inline yuca/ca/continuous.py:771]: `fit_growth` is an old experimental feature that was deprecated and now removed.

File: yuca/ca/continuous.py [hunk 8]
```diff
@@ -945,20 +605,6 @@ def set_params(self, params):
 
         self.no_grad()
         param_start = 0
-        if self.ca_mode != "functional":
```
Comment (riveSunder) [inline yuca/ca/continuous.py:948]: neural functionality moved to its own class `NCA`

File: yuca/ca/continuous.py [hunk 9]
```diff
@@ -1007,70 +653,4 @@ def no_grad(self):
         
 
 if __name__ == "__main__":
-
```
Comment (riveSunder) [inline yuca/ca/continuous.py:1010]: _ad hoc_ tests removed.

**Review by riveSunder — COMMENTED** [review]
Fix typo in alive mask test and clean up white space in a few areas.

Then OK to merge.

File: testing/yuca/ca/test_cca.py (hunk 2 above)
Comment (riveSunder) [reply testing/yuca/ca/test_cca.py:33]: ✓

File: testing/yuca/test_configs.py (hunk 4 above)
Comment (riveSunder) [reply testing/yuca/test_configs.py:14]: ✓

File: yuca/ca/common.py (hunk 5 above)
Comment (riveSunder) [reply yuca/ca/common.py:37]: ✓
